src/dict.c

```diff
@@ -781,6 +798,16 @@ dictEntry *dictFind(dict *d, const void *key)
     return NULL;
 }
 
+dictEntry *dictFind(dict *d, const void *key)
+{
+    return _dictFindByHash(d,key,NULL);
+}
```

Comment:
We can modify `dictFindByHash()` to get unconditionally a value of hash rather than a pointer:
```suggestion
dictEntry *dictFind(dict *d, const void *key)
{
    uint64_t hash = dictHashKey(d, key, d->useStoredKeyApi);
    return _dictFindByHash(d,key,hash);
}
```